docs/rfcs/1007-frozen-users.md

```diff
@@ -0,0 +1,89 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+Frozen users
+============
+
+Frozen users are users that are not allowed to connect to the parsec server.
+
+This feature is not part of the parsec revocation system, but a way for system administrators to temporarily block users from connecting to the server while waiting for organization administrators to actually revoke them. It is exposed as an HTTP route that requires administration token. This way, it is possible for existing directory services to automatically freeze users that have been removed from the directory.
+
+HTTP `frozen_users` route
+-------------------------
+
+This route is made available as `/administration/organizations/<raw_organization_id>/frozen_users` and requires an administration token.
+
+### `GET` method
+
+The `GET` method is used to list frozen status for all users, identified by their email addresses.
+
+The body of the `GET` response is a JSON structure with the following format:
+
+```json
+[
+    {
+        "user_email": "<user_email_1>",
+        "frozen": true
+    },
+    [...]
+    {
+        "user_email": "<user_email_N>",
+        "frozen": false
+    }
+]
+```
+
+The list contains all users of the organization, including those that are not frozen.
+
+However, it does not contain the email addresses of users that have been revoked.
+
+### `PATCH` method
+
+The `PATCH` method is used to modify this status for a given set of users.
+
+The body of the `PATCH` request must be a JSON structure with the following format:
+
+```json
+{
+    "user_email": "<user_email>",
+    "frozen": true
+}
+```
```

Comment:
Considering GET returns an array of users, I think it should be possible (and useful!) to also PATCH an array of users.

The only extra thing to consider is what happen if one or more users are not found (e.g. invalid email). According to the [IETF RFC 5789](https://www.rfc-editor.org/rfc/rfc5789.html#section-2):

> The server MUST apply the entire set of changes atomically and never
>    provide (e.g., in response to a GET during this operation) a
>    partially modified representation.  If the entire patch document
>    cannot be successfully applied, then the server MUST NOT apply any of
>    the changes.

Reply:
We discussed it @touilleMan and agreed that was not necessary for the moment, I added an extra section in the RFC about this future improvement. Is that ok with you?